### PR TITLE
Vanilla server fallback support

### PR DESCRIPTION
**Pull request by Whitcome**
When the mod is installed and the user is playing on a server without Better Combat support, the mod does not work as it is not supported.
This PR aims to have a gameplay fallback mode if playing on Realms or a server without Better Combat support. I like to play with friends on Vanilla servers pretty often, but I miss the animations in this mod.

When this mod is installed and playing on a server without Better Combat:
- Vanilla combat logic will be used
- Vanilla weapon hitboxes will be used
- Better Combat animations will be used :)
- Better combat "Highlight crosshair" will work. This will use the vanilla hit detection in this mode.
- Some of the other options will work such as the option for rendering arms in first-person mode.
- Other players will be animated client-side
- All logic will be client-side.
- Movement speed will not change while swinging weapon
- Leg animations will not lock while moving since vanilla walking/sprinting is faster
- Dual wielding of weapons will not be available just like vanilla logic :(
- Basically, all vanilla combat logic is kept the same except the client-side animations are now available on vanilla servers.

Any suggestions or improvements appreciated

### CODE REVIEW

**Comment by ZsoltMolnarrr** [comment]
Hello Dave!
I appreciate the effort, unfortunately this branch of the project will not be accepted.
Here are my reasons:
- Goes against the goal of the entire project (consistent user experience, replicating Minecraft dungeons), basically too many important features are lost in client side mode
- Too much change on critical parts of the code (realistically I have no time for reviewing and relearning so much of changed core mechanics)
- Too much _non-self-authored_ code to be maintained (I tend to be slow, maintaining others code, I have no capacity for such this amount of additions)

**Comment by Whitcome** [comment]
Sounds good - thank you for the consideration!

**Comment by Julian-Denis** [comment]
I let you know that your branch (even it is closed) is very usefull for me. I use that fallback feature on my personnal modpack and "update" the mod with this feature for my personnal use. I really like that fallback. Very nice job!
